src/arithmetic.c

```diff
@@ -28,6 +28,20 @@ void bstd_add(bstd_number *lhs, const bstd_number *rhs) {
     free((void*)sum);
 }
 
+void bstd_subtract(bstd_number *lhs, const bstd_number *rhs) {
+    bstd_number rhs_copy = *rhs;
```

Comment:
This is not how copying works, we are violating the "const" promise

Reply:
This is a valid way to copy as long as the struct we copy contains no pointers.

https://stackoverflow.com/questions/9127246/copy-struct-to-struct-in-c

Reply:
You're absolutely right, number does not contain pointers